src/extensions/mods.ts

```diff
@@ -1,55 +1,18 @@
 import electron from "electron";
 import {getConfig} from "../utils";
 
-interface PolicyResult {
-    [key: string]: string[];
-}
-
-const parsePolicy = (policy: string): PolicyResult => {
-    const result: PolicyResult = {};
-    policy.split(";").forEach((directive) => {
-        const [directiveKey, ...directiveValue] = directive.trim().split(/\s+/g);
-        if (directiveKey && !Object.prototype.hasOwnProperty.call(result, directiveKey)) {
-            result[directiveKey] = directiveValue;
-        }
-    });
-    return result;
-};
-
-const stringifyPolicy = (policy: PolicyResult): string =>
-    Object.entries(policy)
-        .filter(([, values]) => values?.length)
-        .map((directive) => directive.flat().join(" "))
-        .join("; ");
-
-const unstrictCSP = async () => {
+const unstrictCSP = () => {
     console.log("Setting up CSP unstricter...");
 
-    const cspAllowAll = ["style-src", "connect-src", "img-src", "font-src", "media-src", "worker-src"];
-
-    const isVencord = await getConfig("mods").then((s) => s.includes("vencord"));
     electron.session.defaultSession.webRequest.onHeadersReceived(({responseHeaders}, done) => {
-        let cspHeaders = responseHeaders!["content-security-policy"];
-
-        if (cspHeaders) {
-            const csp = parsePolicy(cspHeaders[0]);
-
-            for (const directive of cspAllowAll) {
-                csp[directive] = ["*", "blob:", "data:", "'unsafe-inline'"];
-            }
-
-            if (isVencord) {
-                // unpkg and cdnjs are used for QuickCss and some plugins' dependencies (e.g. GifEncoder & APNG for FakeNitro)
-                csp["script-src"] ??= [];
-                csp["script-src"].push("'unsafe-eval'", "https://unpkg.com", "https://cdnjs.cloudflare.com");
-            }
-            // Fix Discord's broken CSP which disallows unsafe-inline due to having a nonce (which they don't even use?)
-            csp["script-src"] = csp["script-src"]?.filter((value) => !value.includes("nonce-"));
+        const cspHeaders = Object.keys(responseHeaders!).filter((name) =>
+          name.toLowerCase().startsWith("content-security-policy")
+        );
 
-            cspHeaders[0] = stringifyPolicy(csp);
-        }
+        for (const header of cspHeaders)
+            delete responseHeaders![header];
```

Comment:
whats the point of this filter into loop, why not just `delete responseHeaders["content-security-policy"]`?

Reply:
Multiple `content-security-policy` headers could exist, though unlikely

Reply:
I mean, this specific method has worked for both CC and shelter flawlessly, so I figured there was no particular cause for reworking the code now

Reply:
there's just content-security-policy, nothing else! If u wanna be safe u can also delete content-security-policy-report-only but Discord doesnt use that

Reply:
fair enough